frame/contracts/src/tests.rs

```diff
@@ -1513,6 +1513,59 @@ fn lazy_removal_works() {
 	});
 }
 
+#[test]
+fn lazy_batch_removal_works() {
+	let (code, hash) = compile_module::<Test>("self_destruct").unwrap();
+	ExtBuilder::default().existential_deposit(50).build().execute_with(|| {
+		let min_balance = <Test as Config>::Currency::minimum_balance();
+		let _ = Balances::deposit_creating(&ALICE, 1000 * min_balance);
+		let mut tries: Vec<child::ChildInfo> = vec![];
+
+		for i in 0..3u8 {
+			assert_ok!(Contracts::instantiate_with_code(
+				Origin::signed(ALICE),
+				min_balance * 100,
+				GAS_LIMIT,
+				None,
+				code.clone(),
+				vec![],
+				vec![i],
+			),);
+
+			let addr = Contracts::contract_address(&ALICE, &hash, &[i]);
+			let info = <ContractInfoOf<Test>>::get(&addr).unwrap();
+			let trie = &info.child_trie_info();
+
+			// Put value into the contracts child trie
+			child::put(trie, &[99], &42);
+
+			// Terminate the contract. Contract info should be gone, but value should be still there
+			// as the lazy removal did not run, yet.
+			assert_ok!(Contracts::call(
+				Origin::signed(ALICE),
+				addr.clone(),
+				0,
+				GAS_LIMIT,
+				None,
+				vec![]
+			));
```

Comment:
How does this end up terminating the contract?

Reply:
If you look at the code in `fixtures/self_destruct.wat` then you should find the information, that by passing empty input to thte contract, it will call `seal_terminate`. And as far as I can understand text wasm format it does so :)

Reply:
Correct. The contract terminates if called with an empty input.